src/coreclr/src/gc/unix/gcenv.unix.cpp

```diff
@@ -361,7 +380,12 @@ bool GCToOSInterface::Initialize()
 
         if (status != 0)
         {
+#if defined(TARGET_OSX) && defined(TARGET_AMD64)
+            g_helperPage = 0;
+            goto g_helperPageErrorRecover;
```

Comment:
A nit - could you please avoid usage of the `goto`? Just putting the pthread_mutex_init stuff into the `else` branch of this `if` would work.

Reply:
Yeah.  I think this was the first time I had used a `goto` in 30 yrs.  It made more sense when we were handling every error.

I'll clean it up in the osx-x64 version of the PR.